cloud-azure/src/main/java/com/sequenceiq/cloudbreak/cloud/azure/AzureResourceConnector.java

```diff
@@ -341,9 +341,11 @@ private void checkTemplateDeployment(List<CloudResourceStatus> result, AzureClie
         Collection<CloudResource> result = new ArrayList<>();
         for (CloudInstance instance : instances) {
             String instanceId = instance.getInstanceId();
-            for (CloudResource resource : resources) {
-                if (instanceId.equalsIgnoreCase(resource.getName()) || instanceId.equalsIgnoreCase(resource.getInstanceId())) {
-                    result.add(resource);
+            if (instanceId != null) {
```

Comment:
Do other cloud platforms handle it differently?

Reply:
yes, they do it in different ways